ddl/ddl.go

```diff
@@ -74,166 +74,166 @@ var (
 	// ErrUnsupportedAddPartition returns for does not support add partitions.
-	ErrUnsupportedAddPartition = terror.ClassDDL.New(codeUnsupportedAddPartition, "unsupported add partitions")
+	ErrUnsupportedAddPartition = terror.ClassDDL.New(mysql.ErrUnsupportedDDLOperation, mysql.MySQLErrName[mysql.ErrUnsupportedDDLOperation]+" add partitions")
 	// ErrUnsupportedCoalescePartition returns for does not support coalesce partitions.
-	ErrUnsupportedCoalescePartition = terror.ClassDDL.New(codeUnsupportedCoalescePartition, "unsupported coalesce partitions")
+	ErrUnsupportedCoalescePartition = terror.ClassDDL.New(mysql.ErrUnsupportedDDLOperation, mysql.MySQLErrName[mysql.ErrUnsupportedDDLOperation]+" coalesce partitions")
 	// ErrGeneratedColumnFunctionIsNotAllowed returns for unsupported functions for generated columns.
-	ErrGeneratedColumnFunctionIsNotAllowed = terror.ClassDDL.New(codeGeneratedColumnFunctionIsNotAllowed, "Expression of generated column '%s' contains a disallowed function.")
+	ErrGeneratedColumnFunctionIsNotAllowed = terror.ClassDDL.New(mysql.ErrGeneratedColumnFunctionIsNotAllowed, mysql.MySQLErrName[mysql.ErrGeneratedColumnFunctionIsNotAllowed])
 	// ErrUnsupportedPartitionByRangeColumns returns for does unsupported partition by range columns.
-	ErrUnsupportedPartitionByRangeColumns = terror.ClassDDL.New(codeUnsupportedPartitionByRangeColumns,
-		"unsupported partition by range columns")
-	errUnsupportedCreatePartition = terror.ClassDDL.New(codeUnsupportedCreatePartition, "unsupported partition type, treat as normal table")
-	errUnsupportedIndexType       = terror.ClassDDL.New(codeUnsupportedIndexType, "unsupported index type")
+	ErrUnsupportedPartitionByRangeColumns = terror.ClassDDL.New(mysql.ErrUnsupportedDDLOperation, mysql.MySQLErrName[mysql.ErrUnsupportedDDLOperation]+
+		" partition by range columns")
+	errUnsupportedCreatePartition = terror.ClassDDL.New(mysql.ErrUnsupportedDDLOperation, mysql.MySQLErrName[mysql.ErrUnsupportedDDLOperation]+" partition type, treat as normal table")
+	errUnsupportedIndexType       = terror.ClassDDL.New(mysql.ErrUnsupportedDDLOperation, mysql.MySQLErrName[mysql.ErrUnsupportedDDLOperation]+" index type")
 
 	// ErrDupKeyName returns for duplicated key name
-	ErrDupKeyName = terror.ClassDDL.New(codeDupKeyName, "duplicate key name")
+	ErrDupKeyName = terror.ClassDDL.New(mysql.ErrDupKeyName, mysql.MySQLErrName[mysql.ErrDupKeyName])
 	// ErrInvalidDBState returns for invalid database state.
-	ErrInvalidDBState = terror.ClassDDL.New(codeInvalidDBState, "invalid database state")
+	ErrInvalidDBState = terror.ClassDDL.New(mysql.ErrInvalidDDLState, fmt.Sprintf(mysql.MySQLErrName[mysql.ErrInvalidDDLState], "database"))
 	// ErrInvalidTableState returns for invalid Table state.
-	ErrInvalidTableState = terror.ClassDDL.New(codeInvalidTableState, "invalid table state")
+	ErrInvalidTableState = terror.ClassDDL.New(mysql.ErrInvalidDDLState, fmt.Sprintf(mysql.MySQLErrName[mysql.ErrInvalidDDLState], "table"))
```

Comment:
Do we need to change the way it is used?
Now we use it like `ErrInvalidTableState.GenWithStack("invalid table state %v", tbInfo.State)`. I think we can use it like `ErrInvalidTableState.GenWithStackByArgs(...)` in this PR.

Reply:
Done, thanks